dashboard/app/models/artist.rb

```diff
@@ -284,17 +284,21 @@ def strip_block(block, toolbox_block, create_for_toolbox)
   end
 
   def strip_toolbox_block(block)
-    strip_block block, true, false
-  end
-
-  def strip_solution_block(block)
-    strip_block block, false, false
+    strip_block block, true
   end
 
   def create_toolbox_block(block)
     strip_block block, false, true
   end
 
+  def blocks_match?(toolbox_block, solution_block)
+    stripped_solution_block = strip_block(solution_block)
+    # Return true for either an exact match, or a *_constant block in the
+    # solution with the corresponding *_constant_dropdown block in the toolbox
```

Comment:
presumably this actually returns true for a `*` block in the solution with a corresponding `*_dropdown` block in the toolbox; turn/move by constants are not the only blocks with a dropdown variant